components/gitpod-protocol/src/ide-service.ts

```diff
@@ -5,10 +5,20 @@
  */
 
 import { Event } from "./util/event";
+import { Disposable } from "./util/disposable";
 
-export type IDEState = 'init' | 'ready';
+export type IDEState = 'init' | 'ready' | 'terminated';
 
 export interface IDEService {
```

Comment:
It's a bit late, I know. But how about we renamed this to `IDEFrontendService`?
Every time I see this interface I expect it to be some service on the backend side of things (this is the first "service" on the frontend - some Gitpod API mixings notwithstanding)

Reply:
actually had the same idea yesterday, thought that it is a rather the shape of `IDEFrontendApplication`